paasta_tools/kubernetes/application/tools.py

```diff
@@ -31,45 +34,57 @@ def is_valid_application(deployment: V1Deployment):
     return is_valid
 
 
-def list_namespaced_deployments(
-    kube_client: KubeClient, namespace: str, **kwargs
-) -> Sequence[DeploymentWrapper]:
-    return [
-        DeploymentWrapper(deployment)
-        for deployment in kube_client.deployments.list_namespaced_deployment(
-            namespace, **kwargs
-        ).items
-        if is_valid_application(deployment)
-    ]
+def list_paasta_managed_deployments(
+    kube_client: KubeClient, **kwargs
+) -> Dict[Tuple[str, str], List[Application]]:
```

Comment:
note for future us: might be nice to create a namedtuple or dataclass for this tuple so that we never misuse it by getting the ordering wrong

Reply:
We do use this (service,instance) tuple a lot....